presto-hive/src/test/java/com/facebook/presto/hive/s3/TestPrestoS3FileSystem.java

```diff
@@ -140,6 +141,22 @@ public void testPathStyleAccess()
         }
     }
 
+    @Test
+    public void testUnderscoreBucket()
+            throws Exception
+    {
+        Configuration config = new Configuration();
+        config.setBoolean(S3_PATH_STYLE_ACCESS, true);
+
+        try (PrestoS3FileSystem fs = new PrestoS3FileSystem()) {
+            MockAmazonS3 s3 = new MockAmazonS3();
+            fs.initialize(new URI("s3n://test-bucket_underscore/"), config);
+            fs.setS3Client(s3);
+            ObjectMetadata objectMetadata = fs.getS3ObjectMetadata(new Path("/test/path"));
+            assertEquals(null, objectMetadata);
```

Comment:
Does this actually test the use case? To me it seems like fs.getS3ObjectMetatada will always return null here (looking at the MockAmazonS3 class).

Does this test fail without the code changes?

Reply:
Yep, thats a good call out. Originally this test would throw and illegal argument exception before my fix. After it threw an S3 access exception so I swapped in that MockS3Client which eroded the original value of the test. I'll fix that now.